src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -121,15 +122,41 @@ protected override void WriteExceptionError(Exception ex)
         /// <param name="subscriptionId"> SubscriptionId passed from the cmdlet</param>
         protected void RegisterBlueprintRp(string subscriptionId)
         {
+            // To save time, we'll check the registration state first before making the register call.
             ResourceManagerClient.SubscriptionId = subscriptionId;
-            var response = ResourceManagerClient.Providers.Register(BlueprintConstants.BlueprintProviderNamespace);
+            Provider provider = ResourceManagerClient.Providers.Get(BlueprintConstants.BlueprintProviderNamespace);
 
-            if (response == null)
+            if (provider.RegistrationState == RegistrationState.Registered)
             {
-                throw new KeyNotFoundException(string.Format(Resources.ResourceProviderRegistrationFailed, BlueprintConstants.BlueprintProviderNamespace));
+                return;
             }
-        }
 
+            // The reason we poll for the registrationState for RP registration is because we'd like to make sure the RP registering
+            // happens before the blueprint creation/assignment request is submitted. This should help alleviate the spike in
+            // blueprint creation/assignment failures due to RP not being registered.
+            const int MaxPoll = 20;
+            int pollCount = 0;
+
+            do
+            {
+                if (pollCount > MaxPoll)
+                {
+                    // We should ideally throw a timeout exception here but we're collecting logs on service side about RP registration failures
+                    // and would like to continue with blueprint/assignment creation flow.
+                    //
+                    // To-Do: Add TimeoutException
+                    break;
+                }
+
+                provider = ResourceManagerClient.Providers.Register(BlueprintConstants
+                    .BlueprintProviderNamespace); // Instead of Get, do Register call again since GET takes its sweet time to return the status.
+
+                TestMockSupport.Delay(TimeSpan.FromSeconds(1));
```

Comment:
I saw this on another RP code, but not sure if this is a supported way to introduce a delay.